packages/ui/react-ui-board/src/Grid/Tile.tsx

```diff
@@ -2,36 +2,86 @@
 //
 
-import React from 'react';
+import { draggable } from '@atlaskit/pragmatic-drag-and-drop/element/adapter';
+import React, { useEffect, useRef, useState } from 'react';
 
+import { invariant } from '@dxos/invariant';
 import { type ThemedClassName } from '@dxos/react-ui';
-import { mx } from '@dxos/react-ui-theme';
+import { Card } from '@dxos/react-ui-stack';
 
 import { useGridContext } from './Grid';
 import { getGridRect } from './geometry';
-import { type HasId, type TileLayout } from './types';
+import { type Position, type HasId, type TileLayout } from './types';
 
-// TODO(burdon): Contains surface like Kanban.
-// TODO(burdon): Drag handles only visible on long hover.
+type DragState = 'idle' | 'dragging';
 
 export type TileProps<T extends HasId = any> = ThemedClassName<{
   item: T;
   layout: TileLayout;
-  onClick?: () => void;
 }>;
 
-export const Tile = ({ classNames, item, layout: { x, y, width = 1, height = 1 }, onClick }: TileProps) => {
-  const { grid } = useGridContext('Tile');
+export const Tile = ({ classNames, item, layout }: TileProps) => {
+  const { grid, zoom, onSelect, onDelete, onMove } = useGridContext(Tile.displayName);
+
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [dragState, setDragState] = useState<DragState>('idle');
+  useEffect(() => {
+    const element = ref.current;
+    invariant(element);
+    return draggable({
+      element,
+      canDrag: () => !zoom,
+      onDragStart: () => {
+        setDragState('dragging');
+      },
+      onDrop: ({
+        location: {
+          current: { dropTargets },
+        },
+      }) => {
+        setDragState('idle');
+        const position = dropTargets[0]?.data.position as Position;
+        if (position) {
+          onMove?.(item.id, position);
+        }
+      },
+    });
+  }, [zoom]);
+
+  // TODO(burdon): Title accessor?
+  const title = item.id;
 
   return (
-    <div
-      className={mx('absolute flex flex-col bg-inputSurface border border-separator rounded-sm shadow', classNames)}
-      style={getGridRect(grid, { x, y, width, height })}
-      onClick={onClick}
+    <Card.Root
+      ref={ref}
+      // TODO(burdon): Root should have no padding by default (leave that to Content?)
+      classNames='absolute p-0'
+      style={getGridRect(grid, layout)}
+      onClick={() => onSelect?.(item.id)}
     >
-      <div>
-        <h1 className='p-4'>{item.id}</h1>
-      </div>
-    </div>
+      {/* TODO(burdon): Should the header (toolbar) be inside Content? If so, why is Content separate from Root? (e.g., rather than a Body). */}
+      <Card.Content classNames='h-full'>
+        <Card.Toolbar>
+          {/* TODO(burdon): How to set disabled? */}
+          <Card.DragHandle toolbarItem />
+          {/* TODO(burdon): Heading has strange padding (makes the Toolbar too tall). */}
+          {/* <Card.Heading classNames='grow truncate'>{title}</Card.Heading> */}
+          <h1 className='grow truncate pli-1'>{title}</h1>
```

Comment:
If `Card.Heading` doesn’t do what you want, I’d still recommend using it and overriding its styles as you need, that way it doesn’t become disconnected, i.e. using an `h1` here will add more tech debt.